src/usernotifications.cs

```diff
@@ -0,0 +1,653 @@
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("categoryIdentifier")]
+		string CategoryIdentifier { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("launchImageName")]
+		string LaunchImageName { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[NullAllowed, Export ("sound", ArgumentSemantic.Copy)]
+		UNNotificationSound Sound { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("subtitle")]
+		string Subtitle { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("threadIdentifier")]
+		string ThreadIdentifier { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("title")]
+		string Title { get; set; }
+
+		[Unavailable (PlatformName.TvOS)]
+		[Export ("userInfo", ArgumentSemantic.Copy)]
+		NSDictionary UserInfo { get; set; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotificationRequest : NSCopying, NSSecureCoding {
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("content", ArgumentSemantic.Copy)]
+		UNNotificationContent Content { get; }
+
+		[NullAllowed, Export ("trigger", ArgumentSemantic.Copy)]
+		UNNotificationTrigger Trigger { get; }
+
+		[Static]
+		[Export ("requestWithIdentifier:content:trigger:")]
+		UNNotificationRequest FromIdentifier (string identifier, UNNotificationContent content, [NullAllowed] UNNotificationTrigger trigger);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotificationResponse : NSCopying, NSSecureCoding {
+		[Internal]
+		[Field ("UNNotificationDefaultActionIdentifier")]
+		NSString DefaultActionIdentifier { get; }
+
+		[Internal]
+		[Field ("UNNotificationDismissActionIdentifier")]
+		NSString DismissActionIdentifier { get; }
+
+		[Export ("notification", ArgumentSemantic.Copy)]
+		UNNotification Notification { get; }
+
+		[Internal] // Replaced with typed ActionIdentifier
+		[Export ("actionIdentifier")]
+		NSString _ActionIdentifier { get; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (UNNotificationResponse))]
+	interface UNTextInputNotificationResponse {
+	    
+		[Export ("userText")]
+	    string UserText { get; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[Unavailable (PlatformName.WatchOS)]
+	[BaseType (typeof (NSObject))]
+	interface UNNotificationServiceExtension {
+
+		[Export ("didReceiveNotificationRequest:withContentHandler:")]
+		void DidReceiveNotificationRequest (UNNotificationRequest request, Action<UNNotificationContent> contentHandler);
+
+		[Export ("serviceExtensionTimeWillExpire")]
+		void ServiceExtensionTimeWillExpire (); // FIXME: Better name?
```

Comment:
`ServiceExtension` prefix is part of the type name, maybe just `TimeWillExpire` ?